Updated the image gradient component documentation per new features recently added (#1542)
o3de/o3de.org: content/docs/user-guide/components/reference/gradients/image-gradient.md

| **Pin Preview to Shape** | Uses the bounding box of a compatible Shape component from the given entity to determine the world size of the preview. If **Constrain to Shape** is `Enabled`, the preview will constrain to the actual shape, not just the shape bounds. | EntityId | Current Entity |
| **Preview Position** | Sets the world location of the preview.<br> <br>*This field is available only if there is no entity selected in **Pin Preview to Shape**.* | Vector3: -Infinity to Infinity | X:`0.0`, Y:`0.0`, Z:`0.0` |
| **Preview Size** | Sets the dimensions of the preview.<br> <br>*This field is available only if there is no entity selected in **Pin Preview to Shape**.* | Vector3: 0.0 to Infinity | X:`1.0`, Y:`1.0`, Z:`1.0` |
| **Constrain to Shape** | If `Enabled`, the gradient preview uses the bounds of the entity selected in **Pin Preview to Shape**.<br> <br>*This field is available only if an entity is selected in **Pin Preview to Shape**.* | Boolean | `Disabled` |
| **Tiling** | Sets the number of times to tile the image horizontally (X) and vertically (Y). | Vector2: 0.01 to Infinity | X: `1.0`, Y: `1.0` |

> | **Image Asset** | Sets the source **AZ::RPI::StreamingImageAsset** to generate this gradient's values. | **AZ::RPI::StreamingImageAsset** | None |

Author: Good call, fixed!

### Advanced properties

The **Advanced** group of properties has an in-line toggle switch that enables/disables the entire group of advanced properties. By default, these are disabled.

| Property | Description | Values | Default |
|-|-|-|-|

> | **Channel To Use** | The channel component of the image to sample from.<br><br>The `Terrarium` option is for an image-based terrain file format as defined here: https://www.mapzen.com/blog/terrain-tile-service/ | `Red`, `Green`, `Blue`, `Alpha`, `Terrarium` | `Red` |

Reviewer: ```suggestion
| **Channel To Use** | The channel component of the image to sample from.<br><br>The `Terrarium` option is for an image-based terrain file format as defined by Mapzen [here](https://www.mapzen.com/blog/terrain-tile-service/) | `Red`, `Green`, `Blue`, `Alpha`, `Terrarium` | `Red` |
```

Author: Very cool, thanks updated!

| **Custom Scale** | Choose a scaling operation to be applied to all image data.<br><br>The `Auto` option will automatically scale based on the min/max values in the image data. | `None`, `Auto`, `Manual` | `None` |
| **Range Minimum** | The minimum value used to scale image data into the `0`-`1` range for the gradient. All image values at or below the minimum will be scaled to `0`.<br> <br>*This field is available only if the **Custom Scale** field is set to `Manual`.* | Float: `0.0` to `1.0`  | `0.0` |
| **Range Maximum** | The maximum value used to scale image data into the `0`-`1` range for the gradient. All image values at or above the maximum will be scaled to `1`.<br> <br>*This field is available only if the **Custom Scale** field is set to `Manual`.* | Float: `0.0` to `1.0`  | `1.0` |

> | **Mip Index** | Specify which mip level index to sample from.<br><br>If you specify a mip level higher than the number of mip levels available in your image, then the highest existing mip index will be used. | Integer: `0` to `AZ::RHI::Limits::Image::MipCountMax` (15) | `0` |

Reviewer: I'm wondering if we need to call out MipCountMax here, will a user need to know this when using this component regularly? 

If you want to keep it in, then I suggest moving the MipCountMax to the description (keeps the values easy to scan). Also, you can just minimize to "MipCountMax" or even "Image::MipCountMax" and link to it on GitHub. Makes it easy to find!

```suggestion
| **Mip Index** | Specify which mip level index to sample from.<br><br>If you specify a mip level higher than the number of mip levels available in your image, then the highest existing mip index will be used (see [`MipCountMax`](https://github.com/o3de/o3de/blob/development/Gems/Atom/RHI/Code/Include/Atom/RHI.Reflect/Limits.h)). | Int: `0` to `15` | `0` |
```

Author: Yeah the user wouldn't really need to know `MipCountMax` by name, I just wanted to give some context for why it would be `15`. I like your idea for linking to it so I will do that!

| **Sampling Type** | The sampling type to be used on the image data. | `Point`, `Bilinear` | `Point` |

### Sampling types

> There are several sampling types to choose from. To illustrate, here are previews of the same source image using the
> different supported sampling types.

Reviewer: ```suggestion
There are several sampling types to choose from. To illustrate, here are previews of the same source image using the different supported sampling types.
```

Author: Updated!

#### Point

`Point` sampling is the default and only samples a single point for each pixel of the source image.

![Image Gradient using point sampling](/images/user-guide/components/reference/gradients/image-gradient-component-point.png)

#### Bilinear

The `Bilinear` filter can be used to smooth out the image by requesting four points in a grid around a requested
pixel and then performing interpolation between the points.

![Image Gradient using bilinear interpolation](/images/user-guide/components/reference/gradients/image-gradient-component-bilinear.png)

## ImageGradientRequestBus

Use the following request functions with the `ImageGradientRequestBus` EBus interface to communicate with Image Gradient components in your game.

| Request Name | Description | Parameter | Return | Scriptable |
|-|-|-|-|-|

> | `GetImageAssetPath` | Returns the path of the **AZ::RPI::StreamingImageAsset** property. | None | Path: String | Yes |

Reviewer: I know this formatting was here before, but code should be in in-line code style, like `AZ::RPI::StreamingImageAsset`. Would help if you can update the remaining ones in this PR too - not a big deal if not. 
```suggestion
| `GetImageAssetPath` | Returns the path of the `AZ::RPI::StreamingImageAsset` property. | None | String | Yes |
```

Author: Sure thing!

| **Preview** | Displays the output gradient of this component after all properties are applied. | | |
| **Pin Preview to Shape** | Sets an entity with a compatible Shape component to use as the bounds of the preview if **Constrain to Shape** is `Enabled`. | EntityId | Current Entity |
| **Preview Position** | Sets the world location of the preview.<br> <br>*This field is available only if there is no entity selected in **Pin Preview to Shape**.* | Vector3: -Infinity to Infinity | X:`0.0`, Y:`0.0`, Z:`0.0` |

> | **Preview Size** | Sets the dimensions of the preview.<br> <br>*This field is available only if an entity is selected in **Pin Preview to Shape** that has a valid shape component.* | Vector3: 0.0 to Infinity | X:`1.0`, Y:`1.0`, Z:`1.0` |

Author: Sounds good, I like those better!

| **Constrain to Shape** | If `Enabled`, the gradient preview uses the bounds of the entity selected in **Pin Preview to Shape**.<br> <br>*This field is available only if an entity is selected in **Pin Preview to Shape**.* | Boolean | `Disabled` |
| **Tiling** | Sets the number of times to tile the image horizontally (X) and vertically (Y). | Vector2: 0.01 to Infinity | X: `1.0`, Y: `1.0` |
| **Image Asset** | Sets the source **AZ::RPI::StreamingImageAsset** to generate this gradient's values. | **AZ::RPI::StreamingImageAsset** | None |

### Advanced properties

The **Advanced** group of properties has an in-line toggle switch that enables/disables the entire group of advanced properties. By default, these are disabled.

| Property | Description | Values | Default |
|-|-|-|-|
| **Channel To Use** | The channel component of the image to sample from.<br><br>The `Terrarium` option is for an image-based terrain file format as defined here: https://www.mapzen.com/blog/terrain-tile-service/ | `Red`, `Green`, `Blue`, `Alpha`, `Terrarium` | `Red` |

> | **Custom Scale** | Choose a scaling operation to be applied to all image data.<br><br>The `Auto` option will automatically scaled based on the min/max values in the image data. | `None`, `Auto`, `Manual` | `None` |

Author: Fixed, thanks!

> | **Range Minimum** | The minimum value for each value of the image data to be scaled against.<br> <br>*This field is available only if the **Custom Scale** field is set to `Manual`.* | Float: `0.0` to `1.0`  | `0.0` |

Author: Fixed!

> | **Range Maximum** | The maximum value for each value of the image data to be scaled against.<br> <br>*This field is available only if the **Custom Scale** field is set to `Manual`.* | Float: `0.0` to `1.0`  | `1.0` |

Author: Fixed as well!

> | **Mip Index** | Specify which mip level index to sample from.<br><br>If you specify a mip level higher than the number of mip levels available in your image, then the lowest level mip index will be used. | Integer: `0` to `AZ::RHI::Limits::Image::MipCountMax` (15) | `0` |

Author: Good point, fixed!
